pkg/streamingpromql/engine_test.go

```diff
@@ -3094,13 +3094,20 @@ func TestCompareVariousMixedMetricsVectorSelectors(t *testing.T) {
 
 	labelsToUse, pointsPerSeries, seriesData := getMixedMetricsForTests(true)
 
+	expressions := []string{}
+
 	// Test each label individually to catch edge cases in with single series
 	labelCombinations := testutils.Combinations(labelsToUse, 1)
+
+	// We tried to have this test with 2 labels, but it was failing due to the inconsistent ordering of prometheus processing matchers that result in multiples series, e.g series{label=~"(c|e)"}.
+	// Prometheus might process series c first or e first which will trigger different validation errors for second and third parameter of double_exponential_smoothing.
```

Comment:
I think it's helpful to explain why it gives a different validation error as you explained to me earlier.
It's not a different argument to the function, it's a different value in the range vector.
That is because the range vector of the series being computed against, values are skipped for the native histograms until it gets to a value where it has a float. That aligns with a different scalar value for the argument and thus gives a different error.

Reply:
Effectively it is the different argument that is passed. But there is a part of how native histogram being skipped too. I added your point here da3a0a3fa567884904f918ab20e58b56ca3b8995.

Reply:
It's not clear to me why, if we might get different errors, we skip comparing the values - shouldn't we skip comparing the errors?

Reply:
When we get validation error, `Value` will be nil and we got panic. We still want to compare the error to ensure the validation is correct. More detail discussion is in this comment: https://github.com/grafana/mimir/pull/10873/files/b1eb614926072c1a01437ddd2e6f53b176459960..aaebaa28df8302ecae7669f0ee65be795718831e#r2015549294

Reply:
If `result.Err` is not `nil`, then shouldn't `result.Value` always be `nil`? 

If so, then I think we can drop `skipValueComparison` and modify `RequireEqualResults` to ensure that both results have the same error (or both don't have an error), and if both have the same error, then both `Value`s should be `nil`.

Reply:
I am still thinking the original check to just return and skip Value comparison if they are nil is the simplest.

But let me attempt to still address this suggestion.

>  to ensure that both results have the same error (or both don't have an error)
We can achieve  just using the existing first check
```
	require.Equal(t, expected.Err, actual.Err)
```	
> and if both have the same error, then both Values should be nil.
```
	if expected.Err != nil && actual.Err != nil && expected.Err.Error() == actual.Err.Error() {
		require.Nil(t, expected.Value)
		require.Nil(t, actual.Value)
		return
	}
```

The proposed change is already applied in the last commit.

Reply:
Note: I can't use `errors.Is(expected.Err, actual.Err)`. It won't check the error as the same although the error string is same.